src/java.base/share/classes/java/lang/Class.java

```diff
@@ -923,6 +923,14 @@ public boolean isSynthetic() {
      *     returns "[[[[[[[I"
      * </pre></blockquote>
      *
+     * @apiNote
+     * This is not a strict inverse of {@link #forName(String) forName(String)}.
+     * Distinct classes which share a common name but have different class
+     * loaders will have the identical name. {@linkplain #isPrimitive()
+     * Primitive types}, {@linkplain #isHidden() hidden} classes, and array
+     * classes whose element type is a hidden class or interface cannot be
+     * discovered by {@code forName}.
+     *
```

Comment:
This is an API note in Class::getName. It seems excessive to be saying that primitive types, hidden classes, ... can't be discovered with Class.forName. It might be better to just drop the second half of this note.

Reply:
I think so too.  The spec describes the result of a hidden class is of the form: `{@code N + '/' + <suffix>}`.  If really needed, we can add in that bullet saying "The result is not a binary name."

Reply:
The 3-arg forName specifies it uses the 'getName' format. It's good to update that to refer to binary names, but that API can find array classes, and array classes aren't quite binary names.